MassSpectrometry/IFilteringParams.cs

```diff
@@ -23,6 +23,8 @@ public interface IFilteringParams
         double? MinimumAllowedIntensityRatioToBasePeakM { get; }
         int? NumberOfPeaksToKeepPerWindow { get; }
         int? NominalWindowWidthDaltons { get; }
+        int? NumberOfWindows { get; }
```

Comment:
Does this need to be nullable? If you didn't want any windows, couldn't it just be 1?

Reply:
I'm not sure to be honest. You can imagine that setting both window width and number of windows could cause a collision. Lot's depends on how downstream programs use filtering params and window helper. In window helper, I have an order of operations 1) if window width defined, do something else 2) if window number defined, else 3) apply filtering to the entire spectrum.  Allowing the values to be null (but handling the null assuming conventional parameters) allows crash-free experience. 
![image](https://user-images.githubusercontent.com/16841846/54296140-570c4e00-4582-11e9-8279-20cfd4dc299f.png)

Reply:
Happy to get your further thoughts on this

Reply:
That makes sense to me. I like to stay away from nullable variables when possible, though, because they're slow and can cause issues with tomls. Can we use <1 instead of null? That will prevent issues with negative values, too. I like the order of operations, but how about moving it from the window helper to the filteringParams constructor, so that it will be universal? If it goes into 1) (window width defined), then it can set the number of windows to 0. If it goes into the number of windows, then it sets the window width to 0.

Reply:
Should windowWidthInDaltons be a double? That would make more sense to me.

Reply:
When using window widths, how do you handle the edges of the spectra? I imagine there will be cases where the experimental spectra range divided by the window width is not an integer.

Reply:
Window width can be double (it gets converted). I was worried about "fractional widths). Like if someone made it 0.01Th or something stupid. I'll have a look.

Reply:
as for edges. Two things. One, increment always begins at zero mz but skips empty windows. So if you pick 100mz window and your spectrum begins at 400, the first bin is 400-499.99. if your spectrum starts @ 350, the first bin is 300-399.99 (and you get  a partial bin). Since almost all spectra have varying width, I defaulted to fixed and reproducible behavior. that way all the widows are comparable accross all the scans. probably should take fractional peaks. So, if you take 100 peaks per bin and you have a short bin (like 50 mz out of 100 mz), we could take only 50 peaks. Last thing is that I add 0.000000001 to the edges so nothing gets clipped.

Reply:
forced to give up on the non nullable b/c IFilteringParams doesn't have a constructor, and collides with FilteringParams..... chasing my tail all over mzlib